src/d3d12/d3d12_staging_buffer.cpp

```diff
@@ -55,6 +54,7 @@ namespace wr::d3d12
 	void UpdateStagingBuffer(StagingBuffer* buffer, void * data, std::uint64_t size, std::uint64_t offset)
 	{
 		memcpy(static_cast<std::uint8_t*>(buffer->m_data) + offset, data, size);
+		memcpy(buffer->m_cpu_address + offset, static_cast<std::uint8_t*>(buffer->m_data) + offset, size);
```

Comment:
2 copies? Haven't looked at what your copying but make sure its actually necessary.

Reply:
One copies to the staging buffer cpu memory, and one to the upload heap buffer. I think we're allocating cpu memory because we originally used updateSubResources, which required it. We should probably get rid of that in general though, it would save a lot of memory.